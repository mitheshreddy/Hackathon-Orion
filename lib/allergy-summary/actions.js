'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.loadAllergySummary = loadAllergySummary;
var LOAD_ALLERGY_SUMMARY = exports.LOAD_ALLERGY_SUMMARY = 'LOAD_ALLERGY_SUMMARY';
var LOAD_ALLERGY_SUMMARY_SUCCEEDED = exports.LOAD_ALLERGY_SUMMARY_SUCCEEDED = 'LOAD_ALLERGY_SUMMARY_SUCCEEDED';
var LOAD_ALLERGY_SUMMARY_FAILED = exports.LOAD_ALLERGY_SUMMARY_FAILED = 'LOAD_ALLERGY_SUMMARY_FAILED';

function loadAllergySummary(service, patientId) {
  return {
    type: LOAD_ALLERGY_SUMMARY,
    data: {
      service: service,
      patientId: patientId
    }
  };
}