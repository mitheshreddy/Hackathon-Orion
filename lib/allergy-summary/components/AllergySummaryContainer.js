'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.sagas = exports.reducers = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactRedux = require('react-redux');

var _actions = require('../actions');

var _reducers = require('../reducers');

var _reducers2 = _interopRequireDefault(_reducers);

var _sagas = require('../sagas');

var _sagas2 = _interopRequireDefault(_sagas);

var _service = require('../service');

var _service2 = _interopRequireDefault(_service);

var _AllergySummaryView = require('./AllergySummaryView');

var _AllergySummaryView2 = _interopRequireDefault(_AllergySummaryView);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var propTypes = {
  allergySummary: _propTypes2.default.string,
  refreshAllergySummary: _propTypes2.default.func.isRequired,
  isLoading: _propTypes2.default.bool,
  isFailed: _propTypes2.default.bool
};

var AllergySummaryContainer = function (_React$Component) {
  _inherits(AllergySummaryContainer, _React$Component);

  function AllergySummaryContainer() {
    _classCallCheck(this, AllergySummaryContainer);

    return _possibleConstructorReturn(this, (AllergySummaryContainer.__proto__ || Object.getPrototypeOf(AllergySummaryContainer)).apply(this, arguments));
  }

  _createClass(AllergySummaryContainer, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      document.addEventListener('orionReactComponentRefresh', this.props.refreshAllergySummary);
      this.props.refreshAllergySummary();
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      document.removeEventListener('orionReactComponentRefresh', this.props.refreshAllergySummary);
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          isFailed = _props.isFailed,
          isLoading = _props.isLoading,
          allergySummary = _props.allergySummary;

      return _react2.default.createElement(_AllergySummaryView2.default, { isFailed: isFailed, isLoading: isLoading, text: allergySummary });
    }
  }]);

  return AllergySummaryContainer;
}(_react2.default.Component);

AllergySummaryContainer.propTypes = propTypes;

var mapStateToProps = function mapStateToProps(state) {
  return {
    allergySummary: state.allergySummaryState.allergySummary,
    isLoading: state.allergySummaryState.isLoading,
    isFailed: state.allergySummaryState.isFailed
  };
};

var mapDispatchToProps = function mapDispatchToProps(dispatch, ownProps) {
  return {
    refreshAllergySummary: function refreshAllergySummary() {
      dispatch((0, _actions.loadAllergySummary)(ownProps.service || _service2.default, ownProps.patientId));
    }
  };
};

exports.default = (0, _reactRedux.connect)(mapStateToProps, mapDispatchToProps)(AllergySummaryContainer);


var reducers = _extends({}, { allergySummaryState: _reducers2.default });
exports.reducers = reducers;


var sagas = _sagas2.default;
exports.sagas = sagas;