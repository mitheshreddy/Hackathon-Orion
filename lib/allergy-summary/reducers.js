'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _actions = require('./actions');

var allergySummaryReducers = function allergySummaryReducers() {
  var state = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {
    isFailed: false,
    isLoading: true,
    allergySummary: undefined
  };
  var action = arguments[1];

  var newState = _extends({}, state);

  switch (action.type) {
    case _actions.LOAD_ALLERGY_SUMMARY:
      newState.isFailed = false;
      newState.isLoading = true;
      return newState;

    case _actions.LOAD_ALLERGY_SUMMARY_SUCCEEDED:
      newState.isFailed = false;
      newState.isLoading = false;
      newState.allergySummary = action.allergySummary;
      return newState;

    case _actions.LOAD_ALLERGY_SUMMARY_FAILED:
      newState.isFailed = true;
      newState.isLoading = false;
      return newState;

    default:
      return state;
  }
};

exports.default = allergySummaryReducers;