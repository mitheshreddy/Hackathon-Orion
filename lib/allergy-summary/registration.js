'use strict';

var _reactOnRails = require('react-on-rails');

var _reactOnRails2 = _interopRequireDefault(_reactOnRails);

var _AllergySummaryApplication = require('./components/AllergySummaryApplication');

var _AllergySummaryApplication2 = _interopRequireDefault(_AllergySummaryApplication);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

_reactOnRails2.default.register({
  AllergySummaryApplication: _AllergySummaryApplication2.default
});