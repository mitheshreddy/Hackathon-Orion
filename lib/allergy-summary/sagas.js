'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.fetchAllergySummary = undefined;

var _effects = require('redux-saga/effects');

var _actions = require('./actions');

var _marked = /*#__PURE__*/regeneratorRuntime.mark(fetchAllergySummary),
    _marked2 = /*#__PURE__*/regeneratorRuntime.mark(fetchAllergySummarySaga);

function fetchAllergySummary(action) {
  var allergySummary;
  return regeneratorRuntime.wrap(function fetchAllergySummary$(_context) {
    while (1) {
      switch (_context.prev = _context.next) {
        case 0:
          _context.prev = 0;
          _context.next = 3;
          return (0, _effects.call)(action.data.service.getAllergySummary, action.data.patientId);

        case 3:
          allergySummary = _context.sent;
          _context.next = 6;
          return (0, _effects.put)({ type: _actions.LOAD_ALLERGY_SUMMARY_SUCCEEDED, allergySummary: allergySummary });

        case 6:
          _context.next = 12;
          break;

        case 8:
          _context.prev = 8;
          _context.t0 = _context['catch'](0);
          _context.next = 12;
          return (0, _effects.put)({ type: _actions.LOAD_ALLERGY_SUMMARY_FAILED });

        case 12:
        case 'end':
          return _context.stop();
      }
    }
  }, _marked, this, [[0, 8]]);
}

function fetchAllergySummarySaga() {
  return regeneratorRuntime.wrap(function fetchAllergySummarySaga$(_context2) {
    while (1) {
      switch (_context2.prev = _context2.next) {
        case 0:
          _context2.next = 2;
          return (0, _effects.takeLatest)(_actions.LOAD_ALLERGY_SUMMARY, fetchAllergySummary);

        case 2:
        case 'end':
          return _context2.stop();
      }
    }
  }, _marked2, this);
}

exports.fetchAllergySummary = fetchAllergySummary;
exports.default = [fetchAllergySummarySaga];