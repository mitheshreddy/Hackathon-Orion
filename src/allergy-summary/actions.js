export const LOAD_ALLERGY_SUMMARY = 'LOAD_ALLERGY_SUMMARY';
export const LOAD_ALLERGY_SUMMARY_SUCCEEDED = 'LOAD_ALLERGY_SUMMARY_SUCCEEDED';
export const LOAD_ALLERGY_SUMMARY_FAILED = 'LOAD_ALLERGY_SUMMARY_FAILED';

export function loadAllergySummary(service, patientId) {
  return {
    type: LOAD_ALLERGY_SUMMARY,
    data: {
      service,
      patientId,
    },
  };
}
