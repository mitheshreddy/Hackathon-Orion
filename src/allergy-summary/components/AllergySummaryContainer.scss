body {
    background-color: #c4d7ff;
}

.title {
    text-align: center;
    color: #313e85;
}

.Location {
    font-weight: bold;
    color: #313e85;
}

.Search {
    text-align: center;
    font-weight: bold;
}

.frontpage {
    display: block;
    width: 100%;
    text-align: center;
}

.kaiju-Workspace {
    min-height: 450px;
    background-color: #c4d7ff;
}

.panelcontent {
    position: relative;
}

.apptform {
    color: #313e85;
    font-weight: bold;
    text-align: center;
}