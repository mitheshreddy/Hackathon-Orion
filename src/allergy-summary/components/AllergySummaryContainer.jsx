import React from 'react';
import PropTypes from 'prop-types';

import { connect } from 'react-redux';
import SlidePanel from 'terra-slide-panel';

import { loadAllergySummary } from '../actions';
import allergySummaryReducers from '../reducers';
import allergySummarySagas from '../sagas';
import allergySummaryService from '../service';

import AllergySummaryView from './AllergySummaryView';
import Displayform from './Displayform';
import SearchPhy from './SearchPhy';
import './AllergySummaryContainer.scss';

const propTypes = {
  allergySummary: PropTypes.string,
  refreshAllergySummary: PropTypes.func.isRequired,
  isLoading: PropTypes.bool,
  isFailed: PropTypes.bool,
};

class AllergySummaryContainer extends React.Component {
  constructor() {
    super();
    this.state = ({ isOpen: false, isVisible: false });
    this.handleOnClick = this.handleOnClick.bind(this);
    this.handleSearchClick = this.handleSearchClick.bind(this);
  }
  componentDidMount() {
    document.addEventListener('orionReactComponentRefresh', this.props.refreshAllergySummary);
    this.props.refreshAllergySummary();
  }

  componentWillUnmount() {
    document.removeEventListener('orionReactComponentRefresh', this.props.refreshAllergySummary);
  }

  handleOnClick() {
    this.setState({ isOpen: !this.state.isOpen });
  }

  handleSearchClick() {
    this.setState({ isVisible: !this.state.isVisible });
  }

  render() {
    const { isFailed, isLoading, allergySummary } = this.props;
    const content = this.state.isVisible ? (
    <div className="slide-panel">
      <SlidePanel
        mainContent={<AllergySummaryView isFailed={isFailed} isLoading={isLoading} text={allergySummary} clickHandler={this.handleOnClick} />}
        panelContent={<Displayform />}
        panelBehavior="overlay"
        panelPosition="end"
        panelSize="small"
        isOpen={this.state.isOpen}
        fill
      />
    </div>) : null;
    return (<div>
      <SearchPhy searchHandler={this.handleSearchClick} />
      {content}
    </div>);
  }
}

AllergySummaryContainer.propTypes = propTypes;

const mapStateToProps = state => ({
  allergySummary: state.allergySummaryState.allergySummary,
  isLoading: state.allergySummaryState.isLoading,
  isFailed: state.allergySummaryState.isFailed,
});

const mapDispatchToProps = (dispatch, ownProps) => ({
  refreshAllergySummary: () => {
    dispatch(loadAllergySummary(ownProps.service || allergySummaryService, ownProps.patientId));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(AllergySummaryContainer);

const reducers = Object.assign({}, { allergySummaryState: allergySummaryReducers });
export { reducers };

const sagas = allergySummarySagas;
export { sagas };
