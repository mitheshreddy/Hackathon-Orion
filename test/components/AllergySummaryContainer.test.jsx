/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import { shallowWithStore } from 'enzyme-redux';
import { createMockStore } from 'redux-test-utils';
import AllergySummaryContainer from '../../src/allergy-summary/components/AllergySummaryContainer';
import { LOAD_ALLERGY_SUMMARY } from '../../src/allergy-summary/actions';

it('should render a AllergySummaryContainer with a populated AllergySummary', () => {
  const testContainer = shallowWithStore(
    <AllergySummaryContainer />,
    createMockStore({
      allergySummaryState: {
        allergySummary: 'AllergySummary',
      },
    }),
  );
  expect(testContainer.first().shallow()).toMatchSnapshot();
});

it('should render a AllergySummaryContainer with a nonpopulated AllergySummary', () => {
  const testContainer = shallowWithStore(
    <AllergySummaryContainer />,
    createMockStore({
      allergySummaryState: {},
    }),
  );
  expect(testContainer.first().shallow()).toMatchSnapshot();
});

it('should render a AllergySummaryContainer with IonAllergiesEngine loading', () => {
  const testContainer = shallowWithStore(
    <AllergySummaryContainer />,
    createMockStore({
      allergySummaryState: {},
    }),
  );
  expect(testContainer.first().shallow()).toMatchSnapshot();
});

it('should render a AllergySummary container with a IonAllergiesEngine failed', () => {
  const testContainer = shallowWithStore(
    <AllergySummaryContainer />,
    createMockStore({
      allergySummaryState: {
        isFailed: true,
      },
    }),
  );
  expect(testContainer.first().shallow()).toMatchSnapshot();
});

it('should call the dispatch method', () => {
  const testStore = createMockStore({
    allergySummaryState: {
    },
  });
  const testContainer = shallowWithStore(
    <AllergySummaryContainer patientId={100} />,
    testStore,
  );

  testContainer.props().refreshAllergySummary();
  expect(testStore.isActionTypeDispatched(LOAD_ALLERGY_SUMMARY)).toBe(true);
});
