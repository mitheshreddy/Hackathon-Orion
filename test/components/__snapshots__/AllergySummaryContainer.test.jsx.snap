exports[`test should render a AllergySummary container with a IonAllergiesEngine failed 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <SearchPhy />,
  "nodes": Array [
    <SearchPhy />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <AllergySummaryContainer
        allergySummary={undefined}
        isFailed={true}
        isLoading={undefined}
        refreshAllergySummary={[Function]} />,
      "_debugID": 15,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": AllergySummaryContainer {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleOnClick": [Function],
        "props": Object {
          "allergySummary": undefined,
          "isFailed": true,
          "isLoading": undefined,
          "refreshAllergySummary": [Function],
        },
        "refs": Object {},
        "state": Object {
          "isOpen": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 8,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <SearchPhy />,
        "_debugID": 16,
        "_renderedOutput": <SearchPhy />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <AllergySummaryContainer
    allergySummary={undefined}
    isFailed={true}
    isLoading={undefined}
    refreshAllergySummary={[Function]} />,
}
`;

exports[`test should render a AllergySummaryContainer with IonAllergiesEngine loading 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <SearchPhy />,
  "nodes": Array [
    <SearchPhy />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <AllergySummaryContainer
        allergySummary={undefined}
        isFailed={undefined}
        isLoading={undefined}
        refreshAllergySummary={[Function]} />,
      "_debugID": 11,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": AllergySummaryContainer {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleOnClick": [Function],
        "props": Object {
          "allergySummary": undefined,
          "isFailed": undefined,
          "isLoading": undefined,
          "refreshAllergySummary": [Function],
        },
        "refs": Object {},
        "state": Object {
          "isOpen": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 6,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <SearchPhy />,
        "_debugID": 12,
        "_renderedOutput": <SearchPhy />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <AllergySummaryContainer
    allergySummary={undefined}
    isFailed={undefined}
    isLoading={undefined}
    refreshAllergySummary={[Function]} />,
}
`;

exports[`test should render a AllergySummaryContainer with a nonpopulated AllergySummary 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <SearchPhy />,
  "nodes": Array [
    <SearchPhy />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <AllergySummaryContainer
        allergySummary={undefined}
        isFailed={undefined}
        isLoading={undefined}
        refreshAllergySummary={[Function]} />,
      "_debugID": 7,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": AllergySummaryContainer {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleOnClick": [Function],
        "props": Object {
          "allergySummary": undefined,
          "isFailed": undefined,
          "isLoading": undefined,
          "refreshAllergySummary": [Function],
        },
        "refs": Object {},
        "state": Object {
          "isOpen": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 4,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <SearchPhy />,
        "_debugID": 8,
        "_renderedOutput": <SearchPhy />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <AllergySummaryContainer
    allergySummary={undefined}
    isFailed={undefined}
    isLoading={undefined}
    refreshAllergySummary={[Function]} />,
}
`;

exports[`test should render a AllergySummaryContainer with a populated AllergySummary 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <SearchPhy />,
  "nodes": Array [
    <SearchPhy />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <AllergySummaryContainer
        allergySummary="AllergySummary"
        isFailed={undefined}
        isLoading={undefined}
        refreshAllergySummary={[Function]} />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": AllergySummaryContainer {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleOnClick": [Function],
        "props": Object {
          "allergySummary": "AllergySummary",
          "isFailed": undefined,
          "isLoading": undefined,
          "refreshAllergySummary": [Function],
        },
        "refs": Object {},
        "state": Object {
          "isOpen": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <SearchPhy />,
        "_debugID": 4,
        "_renderedOutput": <SearchPhy />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <AllergySummaryContainer
    allergySummary="AllergySummary"
    isFailed={undefined}
    isLoading={undefined}
    refreshAllergySummary={[Function]} />,
}
`;
